pkg/3scale/amp/operator/redis_reconciler.go

```diff
@@ -60,25 +60,39 @@ func (r *RedisReconciler) Reconcile() (reconcile.Result, error) {
 		return reconcile.Result{}, err
 	}
 
-	dcMutator := reconcilers.DeploymentConfigMutator(
-		reconcilers.DeploymentConfigImageChangeTriggerMutator,
-		reconcilers.DeploymentConfigContainerResourcesMutator,
-		reconcilers.DeploymentConfigAffinityMutator,
-		reconcilers.DeploymentConfigTolerationsMutator,
-		reconcilers.DeploymentConfigPodTemplateLabelsMutator,
-		reconcilers.DeploymentConfigPriorityClassMutator,
-		reconcilers.DeploymentConfigTopologySpreadConstraintsMutator,
-		reconcilers.DeploymentConfigPodTemplateAnnotationsMutator,
-		// 3scale 2.13 -> 2.14
-		upgrade.Redis6CommandArgsEnv,
+	deploymentMutator := reconcilers.DeploymentMutator(
+		reconcilers.DeploymentContainerResourcesMutator,
+		reconcilers.DeploymentAffinityMutator,
+		reconcilers.DeploymentTolerationsMutator,
+		reconcilers.DeploymentPodTemplateLabelsMutator,
+		reconcilers.DeploymentPriorityClassMutator,
+		reconcilers.DeploymentTopologySpreadConstraintsMutator,
+		reconcilers.DeploymentPodTemplateAnnotationsMutator,
+		reconcilers.DeploymentPodContainerImageMutator,
 	)
-	err = r.ReconcileDeploymentConfig(r.DeploymentConfig(redis), dcMutator)
+	redisDeployment := r.Deployment(redis)
+	err = r.ReconcileDeployment(redisDeployment, deploymentMutator)
 	if err != nil {
 		return reconcile.Result{}, err
 	}
 
-	// redis Service
-	err = r.ReconcileService(r.Service(redis), reconcilers.CreateOnlyMutator)
+	// 3scale 2.14 -> 2.15
```

Comment:
Q
Out of curiosity, if I bump from 2.13 to 2.15, it should still migrate correctly right?

Reply:
Again I'm not sure what the behavior would be, I thought that we only needed to support 2.13 -> 2.14 and then 2.14 -> 2.15. But if that's not the case then I think this change (as well as some others) will need to be reverted.

That being said, the actual migration logic for this PR (DeploymentConfigs to Deployments) should still work and migrate correctly. But I'm not sure about the Redis6 stuff that was required for 2.13 -> 2.14.

Reply:
same as above if we can

Reply:
update same as above